pandas/core/indexes/base.py

```diff
@@ -1214,6 +1219,12 @@ def name(self):
 
     @name.setter
     def name(self, value):
+        if self._no_setting_name:
+            # Used in MultiIndex.levels.
+            raise RuntimeError(
```

Comment:
TypeError instead?

Reply:
I think a RuntimeError is best, since we don't check the type or valuje of `value` checked here.